.github/workflows/release.main.kts

```diff
@@ -0,0 +1,250 @@
+        )
+    ),
+    sourceFile = __FILE__.toPath(),
+    targetFileName = "${__FILE__.name.substringBeforeLast(".main.kts")}.yml"
+) {
+    val GITHUB_TOKEN by Contexts.secrets
+    val SONATYPE_OSS_USER by Contexts.secrets
+    val SONATYPE_OSS_PASSWORD by Contexts.secrets
+    val SIGNING_GPG_PASSWORD by Contexts.secrets
+    val SPOCK_BUILD_CACHE_USERNAME by Contexts.secrets
+    val SPOCK_BUILD_CACHE_PASSWORD by Contexts.secrets
+    val GRADLE_ENTERPRISE_ACCESS_KEY by Contexts.secrets
+
+    val (javaVersions, variants) = getMatrixAxes()
+    val buildAndVerify = job(
+        id = "build-and-verify",
+        name = "Build and Verify",
+        runsOn = RunnerType.Custom(expr("matrix.os")),
+        condition = "${github.repository} == 'spockframework/spock'",
+        _customArguments = mapOf(
+            "strategy" to mapOf(
+                "fail-fast" to false,
+                "matrix" to mapOf(
+                    "os" to listOf("ubuntu-latest"),
+                    "variant" to variants,
+                    "java" to javaVersions,
+                    "exclude" to javaVersions
+                        .filter { it.toInt() >= 17 }
+                        .map { javaVersion ->
+                            mapOf(
+                                "os" to "ubuntu-latest",
+                                "variant" to "2.5",
+                                "java" to javaVersion
+                            )
+                        },
+                    "include" to listOf("windows-latest", "macos-latest")
+                        .flatMap { os -> variants.map { os to it } }
+                        .map { (os, variant) ->
+                            mapOf(
+                                "os" to os,
+                                "variant" to variant,
+                                "java" to javaVersions.first()
+                            )
+                        }
+                )
+            )
+        )
+    ) {
+        uses(
+            name = "Checkout Repository",
+            action = CheckoutV3(
+                // Codecov needs fetch-depth > 1
+                fetchDepth = FetchDepth.Value(2)
+            )
+        )
+        uses(
+            name = "Set up JDKs",
+            action = SetupBuildEnv(
+                additionalJavaVersion = expr("matrix.java")
+            )
+        )
+        uses(
+            name = "Build Spock",
+            action = GradleBuildActionV2(
+                arguments = listOf(
+                    "--no-parallel",
+                    "--stacktrace",
+                    "ghActionsBuild",
+                    """"-Dvariant=${expr("matrix.variant")}"""",
+                    """"-DjavaVersion=${expr("matrix.java")}"""",
+                    "-Dscan.tag.main-build"
+                ).joinToString(" ")
+            ),
+            env = linkedMapOf(
+                "ORG_GRADLE_PROJECT_spockBuildCacheUsername" to expr(SPOCK_BUILD_CACHE_USERNAME),
```

Comment:
wouldn't it make sense to just have the `Pair` as constants, that way we don't have to repeat the mapping everywhere.

Reply:
gone to common